File: backend/schemas/app_config.py
from pydantic import BaseModel, Field
from typing import List, Optional, Any
from schemas.database import DBSchema
from schemas.api import APISchema
from schemas.ui import UISchema
from schemas.auth import AuthSchema
from schemas.business_rules import RulesSchema

class Entity(BaseModel):
    name: str
    description: str

class ArchitectureSchema(BaseModel):
    entities: List[Entity]
    description: str

class AppConfig(BaseModel):
    project_name: str
    intent: str
    architecture: ArchitectureSchema
    db: DBSchema
    api: APISchema
    ui: UISchema
    auth: AuthSchema
    rules: RulesSchema
    json_dump: Optional[str] = None

class PipelineStage(BaseModel):
    name: str
    status: str # waiting, running, success, failed
    output: Optional[Any] = None
    errors: List[str] = []

class PipelineStatus(BaseModel):
    stages: List[PipelineStage]
    current_stage: str
    progress: int

File: frontend/src/types/app-config.ts
export interface DBColumn {
    name: string;
    type: string;
    primary_key: boolean;
    nullable: boolean;
    unique: boolean;
    references?: string;
}

export interface DBTable {
    name: string;
    columns: DBColumn[];
}

export interface DBSchema {
    tables: DBTable[];
    relationships: string[];
}

export interface APIEndpoint {
    method: string;
    path: string;
    summary: string;
    request_body?: any;
    response_body?: any;
    auth_required: boolean;
    validation_rules: string[];
}

export interface APISchema {
    endpoints: APIEndpoint[];
}

export interface UIComponent {
    type: string;
    props: any;
    content?: string;
}

export interface UIPage {
    name: string;
    route: string;
    components: UIComponent[];
}

export interface UISchema {
    pages: UIPage[];
}

export interface Permission {
    action: string;
    resource: string;
}

export interface AuthRole {
    name: string;
    permissions: Permission[];
}

export interface AuthSchema {
    roles: AuthRole[];
    matrix: Record<string, string[]>;
}

export interface RulesSchema {
    rules: any[];
}

export interface AppConfig {
    project_name: string;
    intent: string;
    architecture: {
        entities: any[];
        description: string;
    };
    db: DBSchema;
    api: APISchema;
    ui: UISchema;
    auth: AuthSchema;
    rules: RulesSchema;
}

export interface PipelineStage {
    name: string;
    status: "waiting" | "running" | "success" | "failed";
    output?: any;
    errors: string[];
}

export interface PipelineStatus {
    stages: PipelineStage[];
    current_stage: string;
    progress: number;
}
